citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -190,22 +190,37 @@ public ZoteroUpdateItemsStatuses updateCitations(IUser user, String groupId, Lis
     public ICitation createCitation(IUser user, String groupId, List<String> collectionIds, ICitation citation)
             throws ZoteroConnectionException, ZoteroItemCreationFailedException, GroupDoesNotExistException,
             ZoteroHttpStatusException {
-        Optional<ICitationGroup> groupOptional = groupRepository.findFirstByGroupId(new Long(groupId));
-        if (!groupOptional.isPresent()) {
+        List<ICitationGroup> citationGroups = groupRepository.findByGroupId(new Long(groupId));
+        if (citationGroups.size()==0) {
             throw new GroupDoesNotExistException("Group with id " + groupId + " does not exist.");
         }
 
         ICitation newCitation = zoteroManager.createCitation(user, groupId, collectionIds, citation);
-        return citationStore.save(newCitation);
-
+        citationStore.save(newCitation);
+        
         // mark group outdated, so it'll be updated on the next loading
-//        ICitationGroup group = groupOptional.get();
-//        group.setLastLocallyModifiedOn(OffsetDateTime.now().toString());
-//        groupRepository.save((CitationGroup) group);
-
-//        return newCitation;
+        updateCitationGroup(groupId, false, 0);
+        return newCitation;
     }
 
+    public void updateCitationGroup(String groupId, boolean shouldUpdateFromCitations, long citationsSize) {
```

Comment:
updating the citation group should get fresh data from zotero and use that to update the. group

Reply:
Updating from the zotero is actually being done in getGroupItems method. I am fetching from mongoDB here as there's inconsistency in the group items between the ones in zotero and mongoDB and mongo has the updated data.